### PR TITLE
fix(router): load route config from async instructions

### PR DESCRIPTION
**Pull request by btford**
Fixes internal b/23791558

### CODE REVIEW

File: modules/angular2/test/router/integration/router_link_spec.ts [hunk 1]
```diff
@@ -128,6 +129,31 @@ export function main() {
...
        }));
 
+    it('should generate link hrefs when asynchronously loaded',
+       inject([AsyncTestCompleter], (async) => {
+         compile()
+             .then((_) => router.config([
+               new AsyncRoute({
+                 path: '/child-with-grandchild/...',
+                 loader: parentCmpLoader,
+                 as: 'child-with-grandchild'
```
Comment (IgorMinar) [inline modules/angular2/test/router/integration/router_link_spec.ts]: didn't you want to start following CamelCase name conventions?

**Comment by IgorMinar** [comment]
can you improve the commit message please so that it's easier to review / debug in the future.

otherwise lgtm

File: modules/angular2/test/router/integration/router_link_spec.ts (hunk 1 above)
Comment (btford) [inline modules/angular2/test/router/integration/router_link_spec.ts]: I'm going to make the naming convention enforced. I'd rather do all of the changes in one commit than have them be temporarily inconsistent.

**Comment by btford** [comment]
Will do. Thanks for the prompt review. :+1:

**Comment by btford** [comment]
sent to presubmit :dancers:

**Pull request closed without merging**